enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+   the MCO will generate the default configurations for workload partitioning
+   based off of the `Infrastructure.Status`, and restarts the nodes.
+6. After the cluster is up the NTO will then generate the machine configurations
+   using the information provided in the PerformanceProfile manifest.
+7. The MCO applies the updated workload partitioning configurations and restarts
+   the relevant nodes.
+8. Alice will now have a cluster that has been setup with workload partitioning
+   and the desired workloads pinned to the specified CPUSet in the
+   PerformanceProfile.
+
+Applying CPU Partitioning Size Change
+
+```mermaid
+sequenceDiagram
+Alice->>API: Apply PerformanceProfile Manifest
+API-->>NTO: Apply
+loop Generate
+    NTO->>NTO: Machine Configs
+end
+NTO-->>MCO: Apply Machine Configs
+MCO-->>Node: Configure node
+loop Apply
+    Node->>Node: Set kubelet config
+    Node->>Node: Set crio config
+    Node->>Node: Kubelet advertises cores
+end
+Node-->>MCO: Finished Restart
+MCO-->>NTO: Machine Manifests Applied
+NTO-->>API: PerformanceProfile Applied
+```
+
+- **Alice** is a human user who creates an Openshift cluster.
+- **Installer** is assisted installer that applies the user manifest.
+- **NTO** is the node tuning operator.
+- **MCO** is the machine config operator.
+- **Node** is the kubernetes node.
+
+1. Alice sits down and applies the desired PerformanceProfile with the selected
+   workloads to pin.
+2. The NTO will generate the appropriate machine configs that include the
+   Kubelet config and the CRIO config and apply the machine configs.
+3. Once the MCO applies the configs, the node is restarted and the cluster has
+   been updated with the desired workload pinning.
+4. Alice will now have a cluster that has been setup with workload pinning.
+
+#### Variation [optional]
+
+##### E2E Workflow deployment
+
+This section outlines an end-to-end workflow for deploying a cluster with
+workload partitioning enabled and how pods are correctly scheduled to run on the
+management CPU pool.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+1. User sits down at their computer.
+2. **The user creates a `PerformanceProfile` resource with the desired
+   `isolated` and `reserved` CPUSet with the `cpu.workloads[Infrastructure]`
+   added to the enum list.**
+3. **Alice updates the `Infrastructure` CR status to denote that workload
+   partitioning is turned on.**
+4. The user runs the installer to create the standard manifests, adds their
+   extra manifests from steps 2, then creates the cluster.
+5. The kubelet starts up and finds the configuration file enabling the new
+   feature.
+6. The kubelet advertises `management.workload.openshift.io/cores` extended
+   resources on the node based on the number of CPUs in the host.
+7. The kubelet reads static pod definitions. It replaces the `cpu` requests with
```

Comment:
But there is no cpu partitioning present at this time. What will the pods and cri-o do? Just track the consumption and nothing else?

Reply:
This shouldn't be possible anymore, with the changes and the bootstrap mechanism, the crio/kubelet config will be present before kubelet comes online.